.github/actions/setup-miniconda/action.yml

```diff
@@ -63,31 +70,36 @@ runs:
           curl -fsSL "${MINICONDA_URL}" -o "${MINICONDA_INSTALL_PATH}/miniconda.sh"
           bash "${MINICONDA_INSTALL_PATH}/miniconda.sh" -b -u -p "${MINICONDA_INSTALL_PATH}"
           rm -rf "${MINICONDA_INSTALL_PATH}/miniconda.sh"
+
       - name: Update GitHub path to include miniconda install
         shell: bash
         run: |
           MINICONDA_INSTALL_PATH="${RUNNER_TEMP}/miniconda"
           echo "${MINICONDA_INSTALL_PATH}/bin" >> $GITHUB_PATH
+
       - name: Setup miniconda env cache (with env file)
         id: miniconda-env-cache-env-file
         if: ${{ runner.os }} == 'macOS' && ${{ inputs.environment-file }} != ''
         uses: actions/cache@v2
         with:
           path: ${{ runner.temp }}/conda-python-${{ inputs.python-version }}
           key: miniconda-env-${{ runner.os }}-${{ runner.arch }}-${{ inputs.python-version }}-${{ steps.get-date.outputs.today }}-${{ hashFiles(inputs.environment-file) }}
+
       - name: Setup miniconda env cache (without env file)
         id: miniconda-env-cache
         if: ${{ runner.os }} == 'macOS' && ${{ inputs.environment-file }} == ''
         uses: actions/cache@v2
         with:
           path: ${{ runner.temp }}/conda-python-${{ inputs.python-version }}
           key: miniconda-env-${{ runner.os }}-${{ runner.arch }}-${{ inputs.python-version }}-${{ steps.get-date.outputs.today }}
+
       - name: Setup conda environment with python (v${{ inputs.python-version }})
-        if: steps.miniconda-env-cache-env-file.outputs.cache-hit != 'true' && steps.miniconda-env-cache.outputs.cache-hit != 'true'
+        if: steps.miniconda-env-cache-env-file.outputs.cache-hit != 'true' || steps.miniconda-env-cache.outputs.cache-hit != 'true'
```

Comment:
I realize that the previous logic is wrong. The cache is invalid if either one of them has a cache-miss